src/extension.ts

```diff
@@ -61,9 +61,20 @@ export function escapeString(s: string, languageId: string, eol: string)
 			const isLastLine = idx === arr.length - 1;
 			return `${delimiterQuote}${escaped}${!isLastLine ? '\\n' : ''}${delimiterQuote}${(!isLastLine) ? ' \\' : ''}`;
 		}).join(eol);
-	} else {
-		return s;
-	}
+
+        } else if (languageId === "java") {
+        const escaped = s
+            .replace(/\\/g, "\\\\") // backslashes first
+            .replace(/"/g, "\\\"")
+            .replace(/'/g, "\\'")
+            .replace(/\n/g, "\\n")
+            .replace(/\r/g, "\\r")
+            .replace(/\t/g, "\\t")
+            .replace(/\f/g, "\\f");
+        return escaped.split(/\r?\n/).map((s, idx, arr) => `"${s}${idx !== arr.length - 1 ? '\\n' : ''}"`).join(eol);
```

Comment:
I guess Java does not have "raw" string literals?

Are all those replacements needed? I mean, isn't it enough to escape only backslashes and quote marks (") ?

About special characters like \n or \t, I guess it's questionable what's the better way. Maybe it should be optional. But currently other languages don't escape them...